src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4176,40 +4176,62 @@ .FORWARDHELPCATEGORY Cmdlet
     }
     elseif ($help -ne $null)
     {
-        $customPagerCommand = """"
-        $customPagerCommandArgs = """"
-        $customPagerCommandLine = """"
-
-        if ($env:PAGER)
-        {
-            $customPagerCommandLine = $env:PAGER
-
-            # Split the command line into tokens, respecting quoting.
-            $customPagerCommand, $customPagerCommandArgs = & { Write-Output -- $customPagerCommandLine }
+        $pagerArgs = $null
+        
+        # By default use more on Windows and less on Linux.
+        if ($IsWindows) {
+            $pagerCommand = 'more.com'
+        }
+        else {
+            $pagerCommand = 'less'
+            $pagerArgs = '-Ps""Page %db?B of %D:.\. Press h for help or q to quit\.$""'
+        }
+
+        # Respect PAGER environment variable which allows user to specify a custom pager.
+        # Ignore a pure whitespace PAGER value as that would cause the tokenizer to return 0 tokens.
+        if (![string]::IsNullOrWhitespace($env:PAGER)) {
+            if (Get-Command $env:PAGER -ErrorAction Ignore) {
+                # Entire PAGER value corresponds to a single command.
+                $pagerCommand = $env:PAGER
+            }
+            else {
+                # PAGER value is not a valid command, check if PAGER command and arguments have been specified.
+                # Tokenize the specified $env:PAGER value. Ignore tokenizing errors since any errors may be valid
+                # argument syntax for the paging utility.
+                $errs = $null
+                $tokens = [System.Management.Automation.PSParser]::Tokenize($env:PAGER, [ref]$errs)
```

Comment:
I'd expect that Tokenizer is very fast and more fast than Get-Command. So we could always call it and then Get-Command. This simplify code.
We need look $errs and write useful message to user.
Also we could use try-catch for Get-Command and running the command to write more useful messages to user.

Reply:
I'm not sure we should expose tokenizer errors.  We are only using the tokenizer when piping help output to applications.  The arguments passed to those applications could very well fail to tokenize by PowerShell.  For instance, say an app took an argument like `@{HEAD-1}`, perfectly valide for the app but gens a tokenizer error:
```
Missing '=' operator after key in hash literal.
```
For debugging, I propose we leave the __PSPAGER_ARGS env var so the user can inspect it to see exactly what we are passing for args to custom pager application.

Reply:
> I'm not sure we should expose tokenizer errors.

No expose. I think we _could_ use this to create more useful message to user if it is possible.